File: advent_of_code/kotlin/src/test/kotlin/aoc2015/Day01Test.kt
package aoc2015

import common.Day
import common.DayTestBase
import org.junit.jupiter.api.DynamicTest.dynamicTest
import org.junit.jupiter.api.TestFactory

class Day01Test : DayTestBase<String, Long> {
  override val day: Day<String, Long>
    get() = aoc2015.Day01

  @TestFactory
  fun `test part 1`() = listOf(
    Pair("()", 0L),
    Pair("(()", 1L),
    Pair("())", -1L),
    Pair("(())", 0L),
    Pair("()()", 0L),
    Pair("(((", 3L),
    Pair("(()(()(", 3L),
    Pair("))(", -1L),
    Pair(")))", -3L),
    Pair(")())())", -3L),
  ).map { (input, output) ->
    testPart1Raw(input, output)
  }
}

File: scala/aoc/src/test/scala/aoc2015/Day01Test.scala
package aoc2015

import org.scalatest.matchers.should.Matchers.shouldEqual

class Day01Test extends Aoc2015TestBase:
  override val day = Day01

  describe("part 1 test"):
    List(
      ("(((", 3),
      (")))", -3),
    ).map:
      (input, expected) =>
        it(s"part1($input) = $expected"):
          expected shouldEqual part1(input)

  describe("part 2 test"):
    List(
      (")", 1),
      ("()())", 5)
    ).map:
      (input, expected) =>
        it(s"part2($input) = $expected"):
          expected shouldEqual part2(input)
